esp32.c

```diff
@@ -644,6 +686,68 @@ int esp32_CurrentTime( sqlite3_vfs * vfs, double * result )
 	return SQLITE_OK;
 }
 
+#ifdef DEBUG_IO_STATS
+	int esp32_Read_Stats(sqlite3_file *id, void *buffer, int amount, sqlite3_int64 offset) {
+		esp32_file *file = (esp32_file*) id;
+		file->stats.read_cnt++;
+
+		if (file->stats.read_bytes_min > amount) {
+			file->stats.read_bytes_min = amount;
+		}
+		if (file->stats.read_bytes_max < amount) {
+			file->stats.read_bytes_max = amount;
+		}
+
+		int64_t start = esp_timer_get_time();
+		int rc = esp32_Read(id, buffer, amount, offset);
+		file->stats.read_elapsed_us += esp_timer_get_time() - start;
+
+		if (rc == SQLITE_OK) {
+			file->stats.read_failed++;
+		} else {
+			file->stats.read_bytes += amount;
```

Comment:
Is this the other way around?

```suggestion
		if (rc == SQLITE_OK) {
			file->stats.read_bytes += amount;
		} else {
			file->stats.read_failed++;
```

Reply:
Done